pkg/services/dns/dns.go

```diff
@@ -98,6 +107,24 @@ func (h *dnsHandler) addLocalAnswers(m *dns.Msg, q dns.Question) bool {
 			m.Rcode = dns.RcodeNameError
 			return true
 		}
+		ip, err := h.hostsFile.LookupByHostname(q.Name)
```

Comment:
I'd only do this when `q.Qtype != dns.TypeA`
Looking at the code, it looks like `addLocalAnswers` could start with a 
```
			if q.Qtype != dns.TypeA {
				return false
			}
```
instead of having this check inside a loop?